ui/apps/platform/src/Containers/MainPage/AsyncComponent.tsx

```diff
@@ -1,8 +1,15 @@
-import React, { Component } from 'react';
+import React, { Component, ElementType } from 'react';
 import Loader from 'Components/Loader';
 
+type Props = Record<string, never>;
+type State = {
+    component: ElementType | null;
+};
+
 export default function asyncComponent(importComponent) {
-    class AsyncComponent extends Component {
+    class AsyncComponent extends Component<Props, State> {
+        isComponentMounted: boolean;
```

Comment:
Why is this line needed?

Reply:
Good question. Not obvious because the reason is in the unchanged lines below.

Declaration for `this.isComponentMounted` which is in addition to component state declared above.